pkg/engine2/cli.go

```diff
@@ -256,22 +264,107 @@ func setupProfiling() func() {
 	return func() {}
 }
 
-func (em *EngineMain) RunEngine(cmd *cobra.Command, args []string) error {
+func (em *EngineMain) Run(context *EngineContext) (int, []engine_errs.EngineError) {
+	returnCode := 0
+	var engErrs []engine_errs.EngineError
+
+	zap.S().Info("Running engine")
+	err := em.Engine.Run(context)
+	if err != nil {
+		returnCode = 1
+		if ee, ok := err.(engine_errs.EngineError); ok {
+			engErrs = append(engErrs, ee)
+		} else {
+			engErrs = append(engErrs, engine_errs.InternalError{Err: engine_errs.ErrorsToTree(err)})
+		}
+	}
```

Comment:
could you just add a comment saying that specific errors wont be lumped into errors on the run just so we remember